crates/snp_validation/src/lib.rs

```diff
@@ -287,96 +308,129 @@ impl Snp {
+        );
+
+        // Enarx doesn't support migration to another machine
+        ensure!(
+            !report.body.policy.flags.contains(PolicyFlags::MigrateMA),
+            "snp guest policy migration flag was set"
+        );
+
+        // Only check major.minor, as there isn't a revision or patch version in the attestation.
+        if report.body.policy.abi_major > report.body.current_major {
+            bail!("snp policy has higher abi major than firmware");
+        } else if report.body.policy.abi_minor > report.body.current_minor {
+            bail!("snp policy has higher abi minor than firmware");
         }
 
         // Check reserved fields
         if report.body.rsvd1 != 0 || report.body.rsvd3 != 0 || report.body.rsvd4 != 0 {
-            return Err(anyhow!("snp report reserved fields were set"));
+            bail!("snp report reserved fields were set");
         }
 
         for value in report.body.rsvd2 {
-            if value != 0 {
-                return Err(anyhow!("snp report reserved fields were set"));
-            }
+            ensure!(value == 0, "snp report reserved fields were set");
         }
 
         for value in report.body.rsvd5 {
-            if value != 0 {
-                return Err(anyhow!("snp report reserved fields were set"));
-            }
+            ensure!(value == 0, "snp report reserved fields were set");
         }
 
         for value in report.body.policy.rsvd {
-            if value != 0 {
-                return Err(anyhow!("snp report policy reserved fields were set"));
-            }
+            ensure!(value == 0, "snp report policy reserved fields were set");
         }
 
         for value in report.body.plat_info.rsvd {
-            if value != 0 {
-                return Err(anyhow!("snp report platform_info reserved fields were set"));
-            }
+            ensure!(
+                value == 0,
+                "snp report platform_info reserved fields were set"
+            );
         }
 
         // Check fields not set by Enarx
-        for value in report.body.author_key_digest {
-            if value != 0 {
-                return Err(anyhow!(
-                    "snp report author_key_digest field not set by Enarx"
-                ));
-            }
+        for value in report.body.host_data {
+            ensure!(value == 0, "snp report host_data field not set by Enarx");
         }
 
-        for value in report.body.host_data {
-            if value != 0 {
-                return Err(anyhow!("snp report host_data field not set by Enarx"));
-            }
+        ensure!(
+            report.body.vmpl == 0,
+            "snp report vmpl field not set by Enarx"
+        );
+
+        // Check field set by Enarx
+        for value in report.body.report_id_ma {
+            ensure!(
+                value == 255,
+                "snp report report_id_ma field not the value set by Enarx"
+            );
         }
```

Comment:
```rust
    // The report ID of this guest's migration agent
    pub report_id_ma: [u8; 32],
```    

This field was not set by enarx.. we can't make any predictions about the contents. Apparently it's all 255, if not migration agent was enabled.

Please remove the check.

Reply:
where did you read, that this results in all 255?